base_layer/core/src/block.rs

```diff
@@ -157,4 +166,135 @@ impl From<Transaction> for AggregateBody {
     }
 }
 
+#[derive(Default)]
+pub struct BlockBuilder {
+    pub header: Option<BlockHeader>,
+    pub inputs: Option<Vec<TransactionInput>>,
+    pub outputs: Option<Vec<TransactionOutput>>,
+    pub kernels: Option<Vec<TransactionKernel>>,
+}
+
+impl BlockBuilder {
+    pub fn new() -> BlockBuilder {
+        BlockBuilder {
+            header: None,
+            inputs: None,
+            outputs: None,
+            kernels: None,
+        }
+    }
+
+    /// This function adds a header to the block
+    pub fn with_header(&mut self, header: BlockHeader) -> &Self {
+        self.header = Some(header);
+        self
+    }
+
+    /// This function adds the provided transaction inputs to the block
+    pub fn with_inputs(mut self, inputs: &mut Vec<TransactionInput>) -> Self {
+        let mut temp = self.inputs.unwrap_or(Vec::new());
+        temp.append(inputs);
+        self.inputs = Some(temp);
+        self
+    }
+
+    /// This function adds the provided transaction outputs to the block
+    pub fn with_outputs(mut self, outputs: &mut Vec<TransactionOutput>) -> Self {
+        let mut temp = self.outputs.unwrap_or(Vec::new());
+        temp.append(outputs);
+        self.outputs = Some(temp);
+        self
+    }
+
+    /// This function adds the provided transaction kernels to the block
+    pub fn with_kernels(mut self, kernels: &mut Vec<TransactionKernel>) -> Self {
+        let mut temp = self.kernels.unwrap_or(Vec::new());
+        temp.append(kernels);
+        self.kernels = Some(temp);
+        self
+    }
+
+    /// This functions add the provided transactions to the block
+    pub fn with_transactions(mut self, txs: Vec<Transaction>) -> Self {
+        let mut iter = txs.into_iter();
```

Comment:
`txs` already implements Iterator. No need for this

Reply:
iter borrows, into_iter moves. I want to move the transactions into the block not clone them in

Reply:
Ok